### ISSUE TITLE
Return type mismatch in `getAppTrackingAuthorizationStatus`

### ISSUE THREAD

**Issue by themartorana** [adjust/flutter_sdk#122]
Adjust Flutter lib version 4.36.0, Android version 13 

When calling `getAppTrackingAuthorizationStatus` a string value is returned from the method channel which it tries to load into an expected int var, and throws

```_TypeError (type 'String' is not a subtype of type 'int')```

<img width="500" alt="adjust dart — platform_client 2023-12-11 at 3 27 52 PM" src="https://github.com/adjust/flutter_sdk/assets/21433/b8878818-44ce-4d00-a45a-49102d1bd6f0">

**Comment by AMfaraz** [comment]
The return type mismatch in `getAppTrackingAuthorizationStatus` was due to the expected result being an `int`, while the method originally returned a different type. To resolve this, I used the `int.parse` function to convert the result appropriately.

Example Solution:

```dart
String authStatus = await _channel.invokeMethod("getAppTrackingAuthorizationStatus");
// Resolved code using int.parse
int authenticationStatus= int.parse(authStatus);

I have created a pull request to resolve this

**Comment by ramin-deriv** [comment]
Hi
Any update on this?

**Comment by uerceg** [comment]
Hey guys,

Sorry for the delay on this one. @themartorana thank you for reporting, @AMfaraz & @ramin-deriv thank you for going extra miles in opening the PR as well, really appreciate all the effort. However, I don't think that we will proceed with adding changes from that PR because from what I can see, they are trying to solve the issue in the wrong place. 

Originally reported issue is that invocation of `getAppTrackingAuthorizationStatus` on **Android** platform is throwing the above mentioned error. And indeed, that's the issue. Now while looking a bit more in depth, not just with that method, but the same issue should happen if one invokes `requestTrackingAuthorizationWithCompletionHandler` (again) on Android platform. Dart layer is expecting integers to be sent from the native layer which is true in case of iOS: 

- https://github.com/adjust/flutter_sdk/blob/master/ios/Classes/AdjustSdk.m#L667
- https://github.com/adjust/flutter_sdk/blob/master/ios/Classes/AdjustSdk.m#L672

However, it's not the case in native Android implementation:

- https://github.com/adjust/flutter_sdk/blob/master/android/src/main/java/com/adjust/sdk/flutter/AdjustSdk.java#L1087
- https://github.com/adjust/flutter_sdk/blob/master/android/src/main/java/com/adjust/sdk/flutter/AdjustSdk.java#L1154

As you can see, strings are being returned in Android and that's the reason for this error and why the suggested PR is not the way to go; one doesn't need to change Dart layer to expect string to be returned from the native layer - it should not be happening. The way to go is to make Android native layer to return an invalid integer value in these situations (which is `-1`, like in [native iOS SDK implementation](https://github.com/adjust/ios_sdk/blob/master/Adjust/ADJUtil.m#L1201)).

We'll ship the patch update with this fix shortly. I'll keep you posted.

**Comment by uerceg** [comment]
We have just released [v4.37.1](https://github.com/adjust/flutter_sdk/releases/tag/v4.37.1) update which should hopefully address this issue. If you can give it a shot on your end and let us know if it fixed the issue (or not), that'd be great. Looking forward to hearing back from you.
